tests/integration/test_cholesky.py

```diff
@@ -71,6 +69,36 @@ def test_complex(n):
     assert allclose(c, c_np)
 
 
+@pytest.mark.parametrize("n", SIZES)
+def test_batched_3d(n):
+    batch = 4
+    a = _get_real_symm_posdef(n)
+    np_a = a.__array__()
+    a_batched = num.einsum("i,jk->ijk", np.arange(batch) + 1, a)
+    test_c = num.linalg.cholesky(a_batched)
+    for i in range(batch):
+        correct = np.linalg.cholesky(np_a * (i + 1))
+        test = test_c[i, :]
+        assert allclose(correct, test)
+
+
+@pytest.mark.parametrize("n", (2,))  # SIZES)
+def test_batched_4d(n):
+    batch = 2
+    a = _get_real_symm_posdef(n)
+    np_a = a.__array__()
+
+    outer = np.einsum("i,j->ij", np.arange(batch) + 1, np.arange(batch) + 1)
+
+    a_batched = num.einsum("ij,kl->ijkl", outer, a)
+    test_c = num.linalg.cholesky(a_batched)
+    for i in range(batch):
+        for j in range(batch):
+            correct = np.linalg.cholesky(np_a * (i + 1) * (j + 1))
+            test = test_c[i, j, :]
+            assert allclose(correct, test)
+
+
```

Comment:
Do you want to add a tests there for batched_cholesky to call it on transposed arrays? To make sure you get correct layout?

Also, what is someone calls cholesky on 3d or 4d array with the firs dim having size 0  (slicing) or 1 ?

Reply:
added a test for empty batch dimensions